diff --git a/CMakeLists.txt b/CMakeLists.txt
index 10164bc0..cd87e790 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,6 +71,13 @@ function(generate_binary_resource target outfile)
     set(RESOURCES_WORKING_DIR ${CMAKE_CURRENT_BINARY_DIR}/res)
     set(RESOURCE_FILE ${RESOURCES_WORKING_DIR}/breeze-${target}.qrc)
     set(BINARY_RESOURCE_FILE ${CMAKE_CURRENT_BINARY_DIR}/breeze-${target}.rcc)
+    string(FIND ${target} "dark" position)
+    if("${position}" STREQUAL "-1")
+        set(RESOURCE_PREFIX "/icons/breeze")
+    else()
+        set(RESOURCE_PREFIX "/icons/breeze-dark")
+    endif()
+    message(STATUS "Processing resource with prefix: ${RESOURCE_PREFIX}")
 
     # Use $<IF:$<BOOL:${MSVC}>,PATH,LD_LIBRARY_PATH> instead of ${pathVarName} once CMake 3.8 is minimum
     if(MSVC)
@@ -102,6 +109,7 @@ function(generate_binary_resource target outfile)
     add_custom_command(OUTPUT ${BINARY_RESOURCE_FILE}
         COMMAND ${QT_RCC_EXECUTABLE} --binary
                     -o ${BINARY_RESOURCE_FILE}
+                    -root ${RESOURCE_PREFIX}
                     ${RESOURCE_FILE}
         DEPENDS ${RESOURCE_FILE}
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
