diff --git a/CMakeLists.txt b/CMakeLists.txt
index 763f1d1..934d2c7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,7 +41,7 @@ set_package_properties(Canberra PROPERTIES
     TYPE OPTIONAL)
 if (NOT Canberra_FOUND)
     # This is REQUIRED since you cannot tell CMake "either one of those two optional ones are required"
-    find_package(Phonon4Qt6 4.6.60 NO_MODULE REQUIRED)
+    find_package(Phonon4Qt6 4.6.60 NO_MODULE)
     set_package_properties(Phonon4Qt6 PROPERTIES
         DESCRIPTION "Qt-based audio library"
         PURPOSE "Needed to preview notification sounds when Canberra isn't available")
