diff --git a/core/libs/tags/engine/taggingactionfactory.cpp b/core/libs/tags/engine/taggingactionfactory.cpp
index 7c74a183c3..4aa9293b6c 100644
--- a/core/libs/tags/engine/taggingactionfactory.cpp
+++ b/core/libs/tags/engine/taggingactionfactory.cpp
@@ -34,7 +34,6 @@
 #include "tagscache.h"
 #include "coredbaccess.h"
 #include "coredb.h"
-#include "facetags.h"
 
 namespace Digikam
 {
@@ -241,9 +240,7 @@ QList<TaggingAction> TaggingActionFactory::actions() const
             }
 
             actions << newUnderParent;
-            //Don't need to add top level for face tags
-            if (d->parentTagId != FaceTags::personParentTag())
-                actions << newToplevel;
+            actions << newToplevel;
         }
         else // if (createItemTopLevel && createItemTopLevel->action() == defaultAction)
         {
@@ -394,7 +391,7 @@ QString TaggingActionFactory::suggestedUIString(const TaggingAction& action) con
     }
     else // shallCreateNewTag
     {
-        if (action.parentTagId() && d->parentTagId != FaceTags::personParentTag())
+        if (action.parentTagId())
         {
             return i18nc("Create New Tag <tag name> in <parent tag path>", "Create \"%1\" in %2",
                          action.newTagName(), TagsCache::instance()->tagPath(action.parentTagId(), TagsCache::NoLeadingSlash));
