diff --git a/src/i18n/kcatalog.cpp b/src/i18n/kcatalog.cpp
index 2f895ae..4e414cd 100644
--- a/src/i18n/kcatalog.cpp
+++ b/src/i18n/kcatalog.cpp
@@ -66,7 +66,7 @@ static const int s_langenvMaxlen = 64;
 
 static void copyToLangArr(const QByteArray &lang)
 {
-    const int bytes = std::snprintf(s_langenv, s_langenvMaxlen, "LANGUAGE=%s", lang.constData());
+    const int bytes = snprintf(s_langenv, s_langenvMaxlen, "LANGUAGE=%s", lang.constData());
     if (bytes < 0) {
         qCWarning(KI18N) << "There was an error while writing LANGUAGE environment variable:" << std::strerror(errno);
     } else if (bytes > (s_langenvMaxlen - 1)) { // -1 for the \0 character
